kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/networking/NetworkingPlanner.java

```diff
@@ -101,9 +112,17 @@ static Stream<ClusterIngressNetworkingDefinition> planClusterIngressNetworkingDe
                                 Optional<KafkaService> maybeService = clusterResolutionResult.serviceResolutionResult().maybeReferentResource();
                                 // todo, maybe we should not include the case where the service does not resolve, it's optimistic to assume its using the default node id range
                                 List<NodeIdRanges> nodeIdRanges = maybeService.map(s -> s.getSpec().getNodeIdRanges()).orElse(DEFAULT_NODE_ID_RANGES);
-                                return Stream.of(
-                                        clusterIngressNetworkingDefinition(primary, cluster, maybeIngress.get(), nodeIdRanges,
-                                                ingressResolutionResult.ingress().getTls()));
+                                KafkaProxyIngress ingress = maybeIngress.get();
```

Comment:
maybe its the diff view making it worse but this is starting to feel like a very verbose Lambda and should maybe a be a couple of methods.

Reply:
extracted another method for constructing the Definition for a single ingress/cluster combination